src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Solution.cs

```diff
@@ -0,0 +1,167 @@
+        // TODO: This should pass any remaining arguments to the constructor
+        // TODO: isInternal and detectorId should be integrated in backend, not passed here
+        /// <summary>
+        /// Creates a pre-defined Solution component capable of restarting an App instance
+        /// </summary>
+        public static Solution Restart(string resourceUri, bool isInternal, string detectorId)
+        {
+            var resourceLink = UriUtilities.BuildDetectorLink(resourceUri, detectorId);
+            var instructions = $"[Go To Detector]({resourceLink})\n\n" +
+                $"{SolutionConstants.RestartInstructions}";
+
+            return new Solution("Restart Site", resourceUri, isInternal, ActionType.RestartSite, instructions,
+                new string[] { SolutionConstants.AppRestartDescription }, confirm: true, actionName: "Restart App");
+        }
+
+        /// <summary>
+        /// Creates a pre-defined Solution component capable of restarting an App instance
+        /// </summary>
+        public static Solution Restart(string resourceUri, OperationContext context, string detectorId)
+        {
+            return Restart(resourceUri, context.IsInternalCall, detectorId);
+        }
+
+        // TODO: isInternal and detectorId should be integrated in backend, not passed here
+        /// <summary>
+        /// Creates a pre-defined Solution component capable of updating the App Settings of a resource
+        /// </summary>
+        public static Solution UpdateAppSettings(string resourceUri, bool isInternal, string detectorId,
+            Dictionary<string, object> actionArgs)
+        {
+            var descriptionBuilder = new StringBuilder();
+            descriptionBuilder.AppendLine("Apply the following settings changes:");
+            descriptionBuilder.AppendLine();
+            descriptionBuilder.Append(actionArgs.ToMarkdownList());
+
+            var resourceLink = UriUtilities.BuildDetectorLink(resourceUri, detectorId);
+
+            var instructions = $"[Go To Detector]({resourceLink})\n\n" +
+                $"{SolutionConstants.UpdateSettingsInstructions}\n{actionArgs.ToMarkdownList(new string(' ', 3))}";
+
+            return new Solution("Update App Settings", resourceUri, isInternal, ActionType.UpdateSiteAppSettings,
+                instructions, new string[] { descriptionBuilder.ToString() }, actionArgs);
+        }
+
+        /// <summary>
+        /// Creates a pre-defined Solution component capable of updating the App Settings of a resource
+        /// </summary>
+        public static Solution UpdateAppSettings(string resourceUri, OperationContext context, string detectorId,
+            Dictionary<string, object> actionArgs)
+        {
+            return UpdateAppSettings(resourceUri, context.IsInternalCall, detectorId, actionArgs);
+        }
+    }
+
+    /// <summary>
+    /// Actions that a Solution can perform, such as ARM API requests.
+    /// </summary>
+    public enum ActionType
```

Comment:
Private or internal? I don't find other references outside this file.

Reply:
Detector authors will have to be able to create ActionTypes

Reply:
You need to move it to a separate file.

Reply:
> Detector authors will have to be able to create ActionTypes

You need to move it to a separate file.

Reply:
I originally did that, and the Detector editor was unable to find the Enum namespace. Is there a way of importing it that won't be confusing to Detector authors?

Reply:
You can add the reference [here](https://github.com/Azure/Azure-AppServices-Diagnostics/blob/97977ec0643ca86b6ac40a300ade563c78ef88e9/src/Diagnostics.ModelsAndUtils/ScriptUtilities/ScriptHelper.cs#L25).